pkg/prometheus/applyconfiguration.go

```diff
@@ -70,3 +70,39 @@ func prometheusStatusApplyConfigurationFromPrometheusStatus(status *monitoringv1
 
 	return psac
 }
+
+// ApplyConfigurationFromServiceMonitor updates the ServiceMonitor Status subresource.
+func ApplyConfigurationFromServiceMonitor(sm *monitoringv1.ServiceMonitor) *monitoringv1ac.ServiceMonitorApplyConfiguration {
+	smsac := configResourceStatusApplyConfigurationFromConfigResourceStatus(&sm.Status)
+	return monitoringv1ac.ServiceMonitor(sm.Name, sm.Namespace).
+		WithStatus(smsac)
+}
+
+func configResourceStatusApplyConfigurationFromConfigResourceStatus(status *monitoringv1.ConfigResourceStatus) *monitoringv1ac.ConfigResourceStatusApplyConfiguration {
+	crsac := monitoringv1ac.ConfigResourceStatus()
+
+	for _, binding := range status.Bindings {
+		bg := monitoringv1ac.WorkloadBinding().
+			WithGroup(binding.Group).
+			WithName(binding.Name).
+			WithNamespace(binding.Namespace).
+			WithResource(binding.Resource)
+
+		var conditions []*monitoringv1ac.ConfigResourceConditionApplyConfiguration
+		for _, condition := range binding.Conditions {
+			conditions = append(conditions, monitoringv1ac.ConfigResourceCondition().
+				WithType(condition.Type).
+				WithStatus(condition.Status).
+				WithLastTransitionTime(condition.LastTransitionTime).
+				WithReason(condition.Reason).
+				WithMessage(condition.Message).
+				WithObservedGeneration(condition.ObservedGeneration),
+			)
+		}
```

Comment:
We are looping over the slice of conditions twice: 
1. Once in lines 92-101
2. Implicitly, inside `bg.WithConditions()` over the slice of conditions that we have already created. 

We can just append each condition once directly to the binding object with:
```suggestion
				for _, condition := range binding.Conditions {
			bg.WithConditions(
                monitoringv1ac.ConfigResourceCondition().
				WithType(condition.Type).
				WithStatus(condition.Status).
				WithLastTransitionTime(condition.LastTransitionTime).
				WithReason(condition.Reason).
				WithMessage(condition.Message).
				WithObservedGeneration(condition.ObservedGeneration),
			)
		}
```

Reply:
agree, done the changes